### ISSUE TITLE
Driver program scattering amongst agent servers

### ISSUE THREAD

**Issue by carusyte** [chrislusf/gleam#58]
Each time I run the flow, the /tmp folder in every agent server is populated with a folder named by job id and the driver program in it. Somehow my driver program grows into a 26MB file, and the accumulation of the same file is kind of a problem for me. Is there any way to workaround this?

**Comment by carusyte** [comment]
I've already considered the option of submitting all the datasets to the flow and let it handle the workload, but more often than not the dataset comes in a dynamic fashion and it's kind of inefficient to wait until all the datasets are ready.

**Comment by chrislusf** [comment]
I can add some code to remove the executable in agents.

I do not understand your second comment, seems a different issue.

**Comment by carusyte** [comment]
Sorry about my English skills. What I wanted to say is, perhaps gleam is designed to run less frequently than I'd expected but with larger workload?

**Comment by chrislusf** [comment]
Yes. Larger data is all fine.

**Comment by carusyte** [comment]
I quite often got the following message in master's log:
```
2017/09/02 00:17:46 &{0xc42a40ee00}.SendFileResource(_) = _, EOF
2017/09/02 00:17:46 Failed to send related files: EOF
```
And my driver program in master server was brutally killed. I doubted that this was due to the target agent server can't receive any more driver binary from the master? If I occasionally clean up the space in all my agent servers, this kind of error rarely emerge. Can we reuse the same driver binary all round the agent servers given that the binary hasn't changed since the first copy (maybe using MD5 or other reliable and fast checksum algorithm for comparison) ?
